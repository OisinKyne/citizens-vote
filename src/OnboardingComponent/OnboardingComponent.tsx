import React from "react";
import withRoot from "../withRoot";
import { Grid, Typography } from "@material-ui/core";
import MetaMaskLogo from "../static/ethereum-metamask-chrome.png";

class OnboardingComponent extends React.Component {
  state = { open: true };

  constructor(props: any) {
    super(props);

    if (props.web3Injected === true) {
      console.log("Web3 is injected into this Onboarding object.");
      this.state = { open: false };
    }
  }

  render() {
    if (this.state.open === false) {
      return null;
    }
    return (
      <div className={"onboardingDiv"}>
        <Grid container spacing={3} alignContent={"center"}>
          <Grid item xs={12}>
            <a
              href="https://metamask.io/"
              target="_blank"
              rel="noopener noreferrer"
              className={"metamaskUrl"}
            >
              <img
                src={MetaMaskLogo}
                className={"fullWidthImage"}
                alt="MetaMask Fox Logo"
              />
            </a>
          </Grid>
          <Grid item xs={12}>
            <Typography variant="body1" gutterBottom>
              This application connects to the Blockchain through a Chrome
              Extension called{" "}
              <a
                href="https://metamask.io/"
                target="_blank"
                rel="noopener noreferrer"
              >
                Metamask
              </a>
              . This means this website has no control over your vote, it merely
              facilitates the process. Please install Metamask, or any other
              web3 provider to post messages to the blockchain.
            </Typography>
          </Grid>
        </Grid>
      </div>
    );
  }
}

export default withRoot(OnboardingComponent);
