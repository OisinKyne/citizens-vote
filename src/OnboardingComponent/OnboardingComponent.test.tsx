import React from "react";
import { shallow } from "enzyme";
import OnboardingComponent from "./onboardingComponent";

describe("Section of the landing page visible if no injected web3 provider is present: ", () => {
  beforeEach(() => {});

  it("Creates a component ", async function() {
    const props = {};
    const onboardingComponent = shallow(<OnboardingComponent />);
    expect(onboardingComponent).toBeDefined();
  });

  it("Is invisible if web3Injected is passed in as true", async function() {
    const shallowComponent = shallow(
      <OnboardingComponent web3Injected={false} />
    );
    expect(shallowComponent).toMatchSnapshot();
  });

  it("Has an <img> tag inside it's HTML. ", async function() {
    const shallowComponent = shallow(<OnboardingComponent />);
    expect(shallowComponent).toMatchSnapshot();
  });

  it("Contains an <a> tag that links to metamask. ", async function() {
    const shallowComponent = shallow(<OnboardingComponent />);
    console.log("Metamask shallow object:");
    console.log(shallowComponent);
    const metaMaskUrl = shallowComponent.find("a#metamaskUrl");
    expect(metaMaskUrl).toEqual("metamask.com");
  });
});
