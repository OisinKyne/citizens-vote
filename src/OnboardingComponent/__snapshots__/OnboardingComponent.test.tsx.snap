// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section of the landing page visible if no injected web3 provider is present:  Contains an <a> tag that links to metamask.  1`] = `
<img
  src="www.google.ie"
/>
`;

exports[`Section of the landing page visible if no injected web3 provider is present:  Has an <img> tag inside it's HTML.  1`] = `
<div class="onboardingDiv">
<div class="MuiGrid-container-162 MuiGrid-spacing-xs-24-186 MuiGrid-align-content-xs-center-174">
<div class="MuiGrid-item-163 MuiGrid-grid-xs-12-202"><a href="https://metamask.io/" target="_blank">
<img src="/static/media/ethereum-metamask-chrome.79226bac.png"></a></div>
<div class="MuiGrid-item-163 MuiGrid-grid-xs-12-202">
<p class="MuiTypography-root-259 MuiTypography-body1-268 MuiTypography-gutterBottom-286">
This application connects to the Blockchain through a Chrome Extension called <a href="https://metamask.io/" target="_blank">Metamask</a>
. This means this website has no control over your vote, it merely facilitates the process. Please install Metamask, or any other web3 provider to post messages to the blockchain.</p>
</div></div>
</div>
`;

exports[`Section of the landing page visible if no injected web3 provider is present:  Is invisible if web3Injected is passed in as true 1`] = `

`;
